### PR TITLE
added hw 6.2 solution

### PR DESCRIPTION
**Pull request by programmer28**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,13 +3,37 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int CAPACITY = 10;
+    private static Object[] arrayKeys = new Object[CAPACITY];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: fields should be initialized at constructor

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,13 +3,37 @@
...
+    private static Object[] arrayValues = new Object[CAPACITY];
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < arrayKeys.length; i++) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: this method contains too much checks and returns. try to create a field called size to limit your loop iterations and try to remove redundant checks

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,13 +3,37 @@
...
+            }
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < arrayKeys.length; i++) {
+            if (key == null && arrayKeys[i] == null) {
+                return (V)arrayValues[i];
+            } else if (key != null && key.equals(arrayKeys[i])) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: there is a check for key == null and key != null, so one of them is useless :)

Comment (programmer28) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: I`ve changed it

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (programmer28) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Okay, I`ve made it

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 1 above)
Comment (programmer28) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: I`ve changed that by this way

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,13 +3,38 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int CAPACITY = 10;
+    private int size;
+    private Object[] arrayKeys;
+    private Object[] arrayValues;
+
+    public StorageImpl() {
+        this.size = 0;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/impl/StorageImpl.java]: lets remove all `these` keyws

Comment (programmer28) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Sorry, but I don`t understand for why I must remove that. You`ve recommend me to create a counter named size and initialized all field in constructor. What have I made wrong?

Comment (programmer28) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Okay, I`ve fixed that

**Pull request closed without merging**